File: m01_f.ts
type BasicPrimitive=boolean|number|string;

/**
 * Stores key-value variables.
 * 
 * When called, this returns `getStore()` and `setStore()`:
 * - `getStore(key)`
 *   - gets `value` corresponds to the given key.
 * - `setStore(key, value)`
 *   - sets `key` and `value`, this overwrites the value when it's already set. Then returns a single `{key:value}` set of onject.
 * 
 * Notice each `key` must be `string` and `value` be scalar (that is, boolean, number, string).
 */
function useStore() {
	const store: {[key:string]:BasicPrimitive} = {};

	const getStore = (key: string) => {
		if (key in store) return { [key]: store[key] };
		return undefined;
	};
	const setStore = (key: string, value: BasicPrimitive) => {
		return { [key]: Object.assign(store, { [key]: value })[key] };
	};
	return { getStore, setStore };
}

const { setStore } = useStore();

console.log(setStore("name","Kay"))

File: m01_f.py
from typing import Union
T = Union[bool, int, float, str]


def use_store():
  store: dict[str, T] = {}

  def get_store(key: str):
    return {key: store[key]} if key in store else None

  def set_store(key: str, value: T) -> dict[str, T]:
    store[key] = value
    return {key: store[key]}

  return (get_store, set_store)


(get, set_store) = use_store()

print(set_store("name", "Kay"))
